components/skus/browser/resources/skus_internals.tsx

```diff
@@ -51,6 +64,31 @@ function App() {
     })
   }
 
+  const submitCreateOrderFromReceipt = () => {
+    API.createOrderFromReceipt(
+      createOrderFormData.domain, createOrderFormData.receipt).then((r: any) =>
+    {
+      console.log(r)
+      console.log(r.response)
+      // TODO(bsclifton): show/set order ID here
+    })
+  }
+
+  type FormElement = HTMLInputElement | HTMLTextAreaElement | HTMLSelectElement
+  type BaseType = string | number | React.FormEvent<FormElement>
+  function getOnChangeField<T extends BaseType = BaseType> (key: keyof CreateOrderFromReceiptInputFields) {
+    return function (e: T) {
+      const value = (typeof e === 'string' || typeof e === 'number') ? e : e.currentTarget.value
+      if (createOrderFormData[key] === value) {
+        return
+      }
+      setCreateOrderFormData(data => ({
+        ...data,
+        [key]: value
+      }))
+    }
+  }
```

Comment:
It's a little overkill to have this reusable onChange handler factory, when there're only 2 fields using it. But it'll work...
(Instead we could just have either separate state properties instead of a single `CreateOrderFromReceiptInputFields`, or inline handlers, e.g. `onChange={(e) => setCreateOrderFormData(data => ({...data, domain: e.target.value}))}`).
If we keep it maybe also rename to `getOnChangeCreateOrderFormField`.
I can make these changes tomorrow, but then again it seems ok as is for this page.